src/main/java/at/hannibal2/skyhanni/features/garden/farming/LaneSwitchNotification.kt

```diff
@@ -0,0 +1,140 @@
+    private var distancesUntilSwitch: List<Double> = listOf()
+    private var lastBlocksPerSecond = 0.0
+    private var lastPos = LorenzVec(0, 0, 0)
+    private var lastLaneSwitch = SimpleTimeMark.farPast()
+    private var lastWarning = SimpleTimeMark.farPast()
+    private var lastDistancesUntilSwitch: List<Double> = listOf()
+    private var lastDistance = 0.0
+
+    companion object {
+        private val config get() = GardenAPI.config.laneswitch
+
+        @JvmStatic
+        fun playUserSound() {
+            SoundUtils.createSound(
+                config.notification.sound.notificationSound,
+                config.notification.sound.notificationPitch,
+            ).playSound()
+        }
+    }
+
+    private fun switchPossibleInTime(from: LorenzVec, to: LorenzVec, speed: Double, time: Int): Boolean {
+        return from.distance(to) <= speed * time
+    }
+
+    private fun getFarmBounds(plotIndex: Int, playerPos: LorenzVec, lastPos: LorenzVec): List<LorenzVec>? {
+        if(plots[plotIndex].isBarn() || plotIndex == 12) return null
+        val xVelocity = playerPos.x - lastPos.x
+        val zVelocity = playerPos.z - lastPos.z
+        if (xVelocity.absoluteValue > zVelocity.absoluteValue) {
+            val xValueMin = if (plotIndex - 1 == -1 || !plots[plotIndex - 1].unlocked || plots.indexOf(plots[plotIndex - 1]) == 12)
+                plots[plotIndex].box.minX else plots[plotIndex - plotIndex % 5].box.minX
+            val xValueMax = if ((plotIndex + 1) % 5 == 0 || plotIndex + 1 == 25 || !plots[plotIndex + 1].unlocked || plots.indexOf(plots[plotIndex + 1]) == 12)
+                plots[plotIndex].box.maxX else plots[plotIndex + (4 - (plotIndex % 5))].box.maxX
+
+            return listOf(
+                LorenzVec(xValueMin, playerPos.y, playerPos.z), LorenzVec(xValueMax, playerPos.y, playerPos.z)
+            )
+        } else if (xVelocity.absoluteValue < zVelocity.absoluteValue) {
+            val row = (ceil((plotIndex / 5.0) + 0.2) - 1).toInt()
+            val zValueTop = if (plotIndex - 1 == -1 || plotIndex - 5 < 0 || !plots[plotIndex - 5].unlocked || plots.indexOf(plots[plotIndex - 5]) == 12)
+                plots[plotIndex].box.minZ else plots[plotIndex - (5 * row)].box.minZ
+            val zValueBottom = if (plotIndex + 5 > 24 || !plots[plotIndex + 5].unlocked || plots.indexOf(plots[plotIndex + 5]) == 12)
+                plots[plotIndex].box.maxZ else plots[plotIndex + ((4 - row)) * 5].box.maxZ
+
+            return listOf(
+                LorenzVec(playerPos.x, playerPos.y, zValueTop), LorenzVec(playerPos.x, playerPos.y, zValueBottom)
+            )
+        } else if (xVelocity.absoluteValue == 0.0 && zVelocity.absoluteValue == 0.0) {
+            return null
+        } else {
+            return null
+        }
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!config.enabled || !GardenAPI.inGarden() || !plotsLoaded() || !isFarming()) return
```

Comment:
Move in gardenm plots loaded, and is farming into one function isEnabled. use this function both in lines 114 and 84